Source/WebKit/WebProcess/Network/WebSocketChannel.cpp

```diff
@@ -147,14 +148,15 @@ WebSocketChannel::ConnectStatus WebSocketChannel::connect(const URL& url, const
     std::optional<FrameIdentifier> frameID;
     std::optional<PageIdentifier> pageID;
     StoredCredentialsPolicy storedCredentialsPolicy { StoredCredentialsPolicy::Use };
-    if (auto* frame = m_document ? m_document->frame() : nullptr) {
-        RefPtr mainFrame = m_document->localMainFrame();
+    document = m_document.get();
```

Comment:
Why is this here?

Reply:
@rwlbuis this still needs addressing. With that I think this is good to go, but might be good to let @youennf confirm.

Reply:
I was initially worried that webSocketConnectRequest call may change m_document, maybe through IPC calls etc, but I checked it and it simply builds a specific request, so that line should not be needed, I removed it.